src/resolvers/UserTag/usersAssignedTo.ts

```diff
@@ -0,0 +1,131 @@
+      .populate("userId")
+      .lean();
+
+    if (args.after) {
+      // If args.before is provided, then the first element fetched must coincide to the provided cursor
+      if (
+        allusersAssignedTo!.length === 0 ||
+        allusersAssignedTo![0]._id.toString() !== args.after.toString()
+      ) {
+        throw new errors.InputValidationError(
+          requestContext.translate(INVALID_CURSOR_PROVIDED.MESSAGE),
+          INVALID_CURSOR_PROVIDED.PARAM,
+          INVALID_CURSOR_PROVIDED.CODE
+        );
+      }
+
+      // Remove the object with _id = args.after and set hasPreviousPage as true
+      allusersAssignedTo!.shift();
+      connectionObject.pageInfo.hasPreviousPage = true;
+    }
+
+    if (allusersAssignedTo!.length === 0)
+      // Return the default object if the recieved list is empty
+      return connectionObject;
+
+    // Populate the page pointer variables
+    if (allusersAssignedTo!.length === args.first + 1) {
+      connectionObject.pageInfo.hasNextPage = true;
+      allusersAssignedTo!.pop();
+    }
+  }
+
+  // Backward pagination
+  if (args.last) {
+    // Fetch the users
+    allusersAssignedTo = await TagUser.find({
+      ...(args.before && {
+        _id: {
+          $lte: args.before,
+        },
+      }),
+      tagId: parent._id,
+    })
+      .sort({ _id: -1 })
+      .limit(args.before ? args.last + 2 : args.last + 1)
+      .populate("userId")
+      .lean();
+
+    if (args.before) {
+      // If args.before is provided, then the first element fetched must coincide to the provided cursor
+      if (
+        allusersAssignedTo!.length === 0 ||
+        allusersAssignedTo![0]._id.toString() !== args.before.toString()
+      ) {
+        throw new errors.InputValidationError(
+          requestContext.translate(INVALID_CURSOR_PROVIDED.MESSAGE),
+          INVALID_CURSOR_PROVIDED.PARAM,
+          INVALID_CURSOR_PROVIDED.CODE
+        );
+      }
+
+      // Remove the object with _id = args.before and set hasPreviousPage as true
+      allusersAssignedTo!.shift();
+      connectionObject.pageInfo.hasPreviousPage = true;
+    }
+
+    // Return the default object if the recieved list is empty
+    if (allusersAssignedTo!.length === 0) return connectionObject;
+
+    // Populate the page pointer variables
+    if (allusersAssignedTo!.length === args.last + 1) {
+      connectionObject.pageInfo.hasNextPage = true;
+      allusersAssignedTo!.pop();
+    }
+
+    // Reverse the order of the fetched objects as according to Relay Specification the order of
+    // returned objects must always be ascending on the basis of the cursor used
+    allusersAssignedTo = allusersAssignedTo!.reverse();
+  }
+
+  // Create edges from the fetched objects
+  connectionObject.edges = allusersAssignedTo!.map(
+    (tagUser) =>
+      ({
+        node: tagUser.userId,
+        cursor: tagUser._id.toString(),
+      } as Interface_ConnectionEdge<Interface_User>)
```

Comment:
Is this `as` assertion necessary? Is typescript complaining without it?

One problem could be that `userId` type here is `string` instead of `Interface_User`? You can assert the same thing on `tagUser.userId`, then you won't need to import `Interface_ConnectionEdge`.

Reply:
@xoldyckk 
![image](https://user-images.githubusercontent.com/96648934/227719301-1633bacf-3ed7-459c-8659-57a96eaf813a.png)
This is typescript error that I was facing without the as assertion. I can't seem to resolve the same.

Reply:
@EshaanAgg Your first problem is using `!` to assert that something cannot be null, it should be sparingly used and instead there should be strict type checking which exists at runtime.

And in your function your map function you're not returning anything.

This is wrong:-
```ts
=> 
{
  node: tagUser.userId,
  cursor: tagUser._id
}
```
Do it like this:-
```ts
=> 
( {
  node: tagUser.userId,
  cursor: tagUser._id
} )
```

Reply:
Ahh you are right! Made a typing error here. Fixed the same and typescript stopped complaining. 
Thanks for the help @xoldyckk!
Have pushed the changes in the lastest commit.